specification/customerlockbox/resource-manager/Microsoft.CustomerLockbox/preview/2018-02-28-preview/customerlockbox.json

```diff
@@ -0,0 +1,359 @@
+          "200": {
+            "description": "OK. Successfully retrieved operation list.",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/OperationResult"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Operations_List": {
+            "$ref": "./examples/Operations_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.CustomerLockbox/requests/{requestId}": {
+      "get": {
+        "tags": ["Requests"],
+        "description": "Get Customer Lockbox request",
+        "operationId": "Get_Request",
+        "parameters": [
+          {
+            "name": "requestId",
+            "in": "path",
+            "description": "The Lockbox request ID.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Retrieval of Customer Lockbox request successful.",
+            "schema": {
+              "$ref": "#/definitions/LockboxRequestResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get_Request": {
+            "$ref": "./examples/Get_Request.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerLockbox/requests/{requestId}": {
+      "get": {
+        "tags": ["Requests"],
+        "description": "Get Customer Lockbox request",
+        "operationId": "Get_RequestByResourceGroup",
+        "parameters": [
+          {
+            "name": "requestId",
+            "in": "path",
+            "description": "The Lockbox request ID.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "The Azure Resource Group containing the request ID.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Retrieval of Customer Lockbox request successful.",
+            "schema": {
+              "$ref": "#/definitions/LockboxRequestResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get_RequestByResourceGroup": {
+            "$ref": "./examples/Get_RequestByResourceGroup.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.CustomerLockbox/requests/{requestId}/UpdateApproval": {
```

Comment:
Why is approval modeled as a separate post action. Ideally, the "request" has a state property which can contain values like approved, rejected, pending etc. to update the state, a PATCH on /requests/{id} can be performed. the request body will contain this property as one of the inputs. 
Have you explored this option?

Reply:
Yes we will do this in the next iteration of this swagger.